src/utils/processorUtils.ts

```diff
@@ -0,0 +1,229 @@
+import { ProcessorRequest, ProcessorType, Source } from "@openapi/generated";
+import { EventFilter, FilterType, ProcessorFormData } from "../types/Processor";
+import { isCommaSeparatedFilterType } from "@utils/filterUtils";
+import { JSONSchema7, JSONSchema7Definition } from "json-schema";
+
+/**
+ * Prepare configuration params before reading and writing them.
+ * Its only purpose is to manage secret fields and convert to and from
+ * their masked value (empty object: {})
+ * @param data The configuration params object
+ * @param schema The json schema object
+ * @param mode "read" when retrieving data, "write" when preparing data for a save request
+ * @param [parentKey] Key of the parent configuration parameter used when drilling nested data
+ */
+export const prepareConfigParameters = (
+  data: object,
+  schema: object,
+  mode: "read" | "write",
+  parentKey?: string
+): Source["parameters"] => {
+  const newParams: { [key: string]: unknown } = {};
+  const secretNewValue = (value: unknown): unknown => {
+    if (mode === "read") {
+      return typeof value === "object" ? "" : value;
+    }
+    return value === "" ? {} : value;
+  };
+
+  Object.keys(data).map((key) => {
+    const value = (data as { [key: string]: unknown })[key];
+    if (value !== null) {
+      if (typeof value === "object" && Object.keys(value).length > 0) {
+        newParams[key] = prepareConfigParameters(
+          value,
+          schema,
+          mode,
+          parentKey ? `${parentKey}.${key}` : key
+        );
+      } else {
+        if (schema) {
+          const def = getSchemaDefinition(
+            parentKey ? `${parentKey}.${key}` : key,
+            schema
+          );
+          if (def && isJSONSchema(def)) {
+            const { oneOf } = def;
+            if (
+              oneOf &&
+              oneOf.find(
+                (item) => isJSONSchema(item) && item?.format === "password"
+              ) !== undefined
+            ) {
+              newParams[key] = secretNewValue(value);
+            } else {
+              newParams[key] = value;
+            }
+          } else {
+            newParams[key] = value;
+          }
+        }
+      }
+    }
+  });
+  return newParams;
+};
+
+/**
+ * Prepare processor filters data to be included in a processors API request
+ * Filters with invalid or missing properties are excluded from requests
+ * @param filters The filters produced by the processors form
+ */
+export const prepareFilters: (
```

Comment:
It's nice if you point out code, unrelated to this change, that you've moved :smile:

Reply:
I'll add a review with some comments later to explain the content of `processorUtils`